Update CI to run tests and docs checks for both python 3.12 and 3.13

roryclaydon1994 opened the pull request:
<!--
     For Work In Progress Pull Requests, please use the Draft PR feature,
     see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

     For a timely review/response, please avoid force-pushing additional
     commits if your PR already received reviews or comments.

     Before submitting a Pull Request, please ensure you've done the following:
     - 📖 Read the [EUCLID RAG Contributing Guide](CONTRIBUTING.md)
     - 📖 Read the [EUCLID RAG Code of Conduct](CODE_OF_CONDUCT.md)
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.

     NOTE: Pull Requests from forked repositories will need to be reviewed by
     a EUCLID RAG Team member before any CI builds will run. Once your PR is approved
     with a `/ci` reply to the PR, it will be allowed to run subsequent builds without
     manual approval.
-->

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update
- [x] CI / Dev-Ops

## Description

The current set of CI pipelines are failing due to missing imports amongst other errors. This PR fixes these simple issues, and adds the option to also test on python 3.13 to ensure we can support later python versions. I've also added some simple contributing guides, code of conduct, and a gitlab template.

## Related Tickets & Documents

<!--
For pull requests that relate or close an issue, please include them
below.  We like to follow [Github's guidance on linking issues to pull requests](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

For example having the text: "closes #1234" would connect the current pull
request to issue 1234.  And when we merge the pull request, Github will
automatically close the issue.
-->

- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [ ] Landing page loads in a reasonable time

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 70% and above._

- [ ] Yes
- [x] No, and this is why: this is pure CI
- [ ] I need help with writing tests

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)

roryclaydon1994 commented on pyproject.toml:
```diff
@@ -43,16 +43,17 @@ dynamic = ["version"]
+]
+docs = [
     "documenteer[guide]<2",
     "scriv",
-]
-
+    "sphinx"
+    ]
```
Make sure to propagate to the makefile

roryclaydon1994 replied:
Yes! The docs needed to be added to the pip install in the `init` target of the `Makefile`. That should be propagated now

jeipollack replied:
okay, thanks!

jeipollack commented on .github/CODE_OF_CONDUCT.md:
```diff
@@ -0,0 +1,76 @@
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team at hello@opensauced.pizza. All
```
We have to edit the email address for contacting the project team.

jeipollack replied:
How about replacing with the following?

_Instances of abusive, harassing, or otherwise unacceptable behavior may be
reported by opening a private issue or by contacting a project maintainer via GitHub.
We are in the process of establishing a dedicated contact email._

roryclaydon1994 replied:
Done. Do you think the location of the files makes sense in the `.github/` subdir? I put it there to link easily with the template I added, but perhaps a dedicated folder is more sensible?

jeipollack replied:
By "link easily" do you mean in the README? For WaveDiff, we have it at the top-level. I think I've seen it this way for other repos perhaps done for ease of visibility.

I suppose as long as it's linked to in the README, I think `.github` is fine for now, but I am open to other "sensible" proposals 😄

jeipollack replied:
let's keep it as and when discuss codebase layout post-demo we can revisit these templates.

jeipollack commented on .github/CONTRIBUTING.md:
```diff
@@ -0,0 +1,59 @@
+We actively welcome your pull requests. Linking to an existing issue is preferred.
+
+1. Fork the repo and create your branch from `main` or the designated default branch.
+2. Use descriptive branch names like `feature/add-new-thing`.
```
I like the idea of including the issue number in the branch name, e.g.

`feature/123-add-new-thing` 

What do you think?

roryclaydon1994 replied:
Yes definitely. Updated now.

jeipollack commented on docs/conf.py:
```diff
@@ -36,7 +38,7 @@
 
 
-html_static_path = ["_static"]  # create this folder (even if empty)
+html_static_path = [str(static_path)]  # create this folder (even if empty)
```
While this is okay, I think it was me that created `_static` because it was missing and triggered an error.

 I read that changing from `_static` to `static` breaks Sphinx convention (https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_static_path).
 
 GPT & Co-pilot both state:
 
 Sphinx uses `_static/` as the conventional and expected directory for user-provided static files (e.g., logos, CSS, JS). Renaming it to static/ technically works when referenced consistently in conf.py, but it goes against the standard convention and may lead to:

- Confusion for new contributors who are used to _static/
- Breakage with tools or themes that assume _static/
- Potential compatibility issues with Sphinx updates or extensions

To keep things aligned with Sphinx best practices, I'd recommend:

- Renaming static/ back to _static/
- Updating html_static_path = ["_static"]
- Updating image references in html_theme_options to _static/logo.png

roryclaydon1994 replied:
The change was made because `_static` never existed, only `static`, which is referenced in the `app.py`. We haven't formalised how to best propagate such directories, so I would suggest we wait until the other hackathon PRs are merged then revisit this. If you think that's a good idea, I can make an issue for it?

jeipollack replied:
Yes, good idea. after you open the issue, we can mark this thread resolved.

jeipollack replied:
Nvm I opened the issue #30  the lazy way using this and selecting `reference in new issue`.

Issue has been captured so we can resolve this thread.

roryclaydon1994 commented on pyproject.toml:
```diff
@@ -81,7 +82,7 @@ branch = true
 
 [tool.coverage.paths]
-source = ["python", ".tox/*/site-packages"]
+source = ["src", ".tox/py*/**/site-packages"]
```
No sorry, that was inadvertent. I have reverted it but left the recursive wildcard in

roryclaydon1994 commented:
## Actions complete

Thanks for the thorough review @jeipollack. I've addressed all the comments with the exception of `static` -> `_static` change. The reason for this is that the `static` folder already existed, so what I did here was just make the current set up for sphinx work with our repo structure, not vice versa. Once we get the hackathon work merged, I'm happy to make this change, but changing now is likely to cause conflicts.

jeipollack commented:
Thanks @roryclaydon1994 for addressing the comments in this conversation. I opened issue #30 to discuss whether or not to follow Sphinx convention and we can go over the templates post-demo time.

I am going to go ahead and Merge this PR! 

Thanks for working on this not-so-glamourous but essential work! 👏🏽 🏆
